torch/csrc/autograd/python_engine.cpp

```diff
@@ -31,6 +32,10 @@ static Engine& get_python_engine() {
 
 namespace torch { namespace autograd { namespace python {
 
+PythonEngine::PythonEngine () {
+  max_recursion_depth_ = 0.1*Py_GetRecursionLimit();
```

Comment:
A little note explaining how we got the magic number 0.1 would be appreciated here :)

Reply:
So, I guess, now the constructor for PythonEngine must be GIL protected. Have we checked this invariant is respected in all the use sites?